### ISSUE TITLE
 Change POST request to url

### ISSUE THREAD

**Issue by Alenorze-SoulSinger** [ronzeidman/ng2-ui-auth#142]
![image](https://user-images.githubusercontent.com/36929871/38267965-9c7ec41a-3785-11e8-9ea5-0cdff395d901.png)

satellizer had such requests, but here they are slightly different

![image](https://user-images.githubusercontent.com/36929871/38267970-a054532a-3785-11e8-95a0-005746de0276.png)

Is it possible to bring them to the first value?

**Comment by ronzeidman** [comment]
In previous versions they were the same but it was hard to understand what was actually sent to the server. I changed it to be this way in the latest version. `authorizationData` is the actual data that was sent to the authorization endpoint, `oauthData` is the data received from the authorization endpoint and `userData` is the additional data you've specified. This is different from the original satellizer that did some kind of mege of everything but I think it's better organized and more predictable.

**Comment by Alenorze-SoulSinger** [comment]
Yes, this is the right decision, but can tell me how to bring these headlines to that value, just on the backend this module and it does not work for another

**Comment by ronzeidman** [comment]
You can merge all of the object to a single one, but there will also be some name changes (snake-case to camel-case in some variables and the "scope" variable is concatinated in the authorizationData in the way defined by the provider settings.
first try doing something like:
```js
Object.keys(req.body).reduce((acc, key) => ({...acc, ...req.body[key]}), {}) //this will merge every key to every value;
```
or if you want a less general and more understandable code
```js
{ ...req.body.authorizationData, ...req.body.oauthData, ...req.body.userData }
```
if you have missing properties or properties in the wrong shape after that you'll have to try and fix each one of them manually.

**Comment by navneet35371** [comment]
@ronzeidman Hey, I can see your decision to semantically distribute the keys but it breaks some of the libraries out there. Like django-rest-auth which tries to fetch from request body directly.

https://github.com/Tivix/django-rest-auth/blob/master/rest_auth/registration/serializers.py#L85

Anyways, Can you explain the above solution a bit? or a example app maybe?

**Comment by ronzeidman** [comment]
@navneet35371 I see what you mean, maybe it should be configurable. Anyway in the meantime can you add a simple middleware for the routes handled by the django-rest-auth library to merge the body's properties (probably you only need to merge the oauthData with the userData)
